src/maptm/src/maptm_config.c

```diff
@@ -0,0 +1,519 @@
+    if (!((IPV4_ADDRESS_SIZE-mapt_rule->prefix4len) == (mapt_rule->ealen)))
+    {
+        mapt_rule->domaine_pssid = ipv6addr&(~( suffix  << (mapt_rule->ealen - (IPV4_ADDRESS_SIZE - mapt_rule->prefix4len))));
+    }
+    
+    // Set Public IPv4 Address
+    uint32_t swapped_suffix = ((suffix>>24)&0xff) 
+        | ((suffix<<8)&0xff0000) 
+        | ((suffix>>8)&0xff00) 
+        | ((suffix<<24)&0xff000000);
+        
+    struct in_addr ipv4BinPrefixRule;
+    inet_pton(AF_INET, mapt_rule->ipv4prefix, &ipv4BinPrefixRule);
+    struct in_addr ipv4BinPublicAddress;
+    ipv4BinPublicAddress.s_addr = ((uint32_t)ipv4BinPrefixRule.s_addr | swapped_suffix ); 
+    inet_ntop(AF_INET, &ipv4BinPublicAddress, mapt_rule->ipv4PublicAddress, IPV4_ADDRESS_SIZE);
+    
+}
+
+// Get MAP-T configuration
+struct mapt* maptm_getconfigure(char *option95, char *iapd, int iapd_len)
+{
+    struct mapt *selected_rule = get_Mapt_Rule(option95, iapd);
+    if (selected_rule == NULL) return NULL;
+
+    configureMapDomain(iapd, iapd_len, selected_rule);
+    
+    return selected_rule;
+}
+
+// MAP-T Firewall configuration
+bool maptm_ovsdb_nfm_set_rule(const char *rule, bool enable)
+{
+    struct schema_Netfilter set;
+    json_t *where = NULL;
+    int rc = 0;
+
+    memset(&set, 0, sizeof(set));
+    set._partial_update = true;
+    SCHEMA_SET_INT(set.enable, enable);
+
+    where = ovsdb_where_simple(SCHEMA_COLUMN(Netfilter, name), rule);
+    
+    if (!where)
+    {
+        LOGE("[%s] Set NAT Netfilter rule: create filter failed", rule);
+        return false;
+    }
+
+    rc = ovsdb_table_update_where(&table_Netfilter, where, &set);
+    if (rc != 1)
+    {
+        LOGE("[%s] Set NAT Netfilter rule: unexpected result count %d", rule, rc);
+        return false;
+    }
+    return true;
+}
+
+// Set MAP-T firewall rules
+bool maptm_ovsdb_nfm_rules(bool enable)
+{
+    LOGT("%s, config Firewall MAP-T",__func__);
+    
+    if (!maptm_ovsdb_nfm_set_rule(V4_CHCEK_FORWARD, !enable)) return false;
+    if (!maptm_ovsdb_nfm_set_rule(V4_MAPT_CHCEK_FORWARD, enable)) return false;
+    if (!maptm_ovsdb_nfm_set_rule(V4_MAPT_TCP_CHCEK_1, enable)) return false;
+    if (!maptm_ovsdb_nfm_set_rule(V4_MAPT_TCP_CHCEK_2, enable)) return false;
+    if (!maptm_ovsdb_nfm_set_rule(V6_MAPT_TCP_CHCEK_3, !enable)) return false;
+
+    return true;
+}
+
+// Stop MAP-T functionality
+bool stop_mapt(void)
+{
+    if (!maptm_ovsdb_nfm_rules(false)) LOGE("Could not disable Firewall");
+    if (!osn_mapt_stop()) return false;
+    
+    LOGD("Stopped Mapt");    
+    return true;
+}
+
+// Configure MAP-T functionality
+bool config_mapt(void)
+{
+    if (strucWanConfig.mapt_95_value[0] == '\0' ||  strucWanConfig.iapd[0] == '\0')
+    {
+        LOGE("Unable to configure MAP-T option");
+        return false;
+    }
+    
+    /* Hard Coded value: Must be changed */
+    char subnetcidr4[20] = "192.168.1.1/24";
```

Comment:
The comment says it all.

Reply:
Eventually fixed in: https://github.com/plume-design/opensync/pull/37/files#diff-f821934fbe65584ec05d12411d932d3e4be3451cda3a027db94568724848af03R484